sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -637,4 +637,39 @@ object IntervalUtils {
 
     new CalendarInterval(totalMonths, totalDays, micros)
   }
+
+  /**
+   * Adjust interval so 30-day time periods are represented as months
+   */
+  def justifyDays(interval: CalendarInterval): CalendarInterval = {
+    val months = Math.addExact(interval.months, interval.days / DAYS_PER_MONTH.toInt)
```

Comment:
if `days` is negative, will we end like `-1 month 2 days`?

Reply:
good catch, we shall calculate all days first just like other 2 funcs